uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -92,6 +93,25 @@ define([
       section.dataIndex[leaveRequest.id] = leaveRequest;
     }
 
+    /**
+     * Attaches the entitlement information to the absence type it belongs to
+     * If there is no entitlement for a given absence type, a default
+     * entitlement object is assigned instead
+     */
+    function attachEntitlementsToAbsenceTypes () {
+      vm.absenceTypes = vm.absenceTypes.map(function (absenceType) {
+        var entitlement = _.find(vm.entitlements, function (entitlement) {
```

Comment:
This would run a little faster:

```js
function attachEntitlementsToAbsenceTypes () {
  var entitlements = _.indexBy(vm.entitlements, 'type_id');

  vm.absenceTypes = vm.absenceTypes.map(function (absenceType) {
    var entitlement = entitlements[absenceType.id];

     // ...
  });
}
```